uaclient/tests/test_config.py

```diff
@@ -0,0 +1,172 @@
+        tmp_subdir = self.tmp_path('does/not/exist')
+        cfg = UAConfig({'data_dir': tmp_subdir})
+
+        assert False is os.path.isdir(tmp_subdir), (
+            'Found unexpected directory %s' % tmp_subdir)
+        assert None is cfg.write_cache('somekey', 'someval')
+        assert True is os.path.isdir(tmp_subdir), (
+            'Missing expected directory %s' % tmp_subdir)
+        assert 'someval' == cfg.read_cache('somekey')
+
+    @params(('dictkey', {'1': 'v1'}), ('listkey', [1, 2, 3]))
+    def test_write_cache_writes_json_string_when_content_not_a_string(
+            self, key, value):
+        """When content is not a string, write a json string."""
+        tmp_dir = self.tmp_dir()
+        cfg = UAConfig({'data_dir': tmp_dir})
+
+        expected_json_content = json.dumps(value)
+        assert None is cfg.write_cache(key, value)
+        with open(self.tmp_path(key, tmp_dir), 'r') as stream:
+            assert expected_json_content == stream.read()
+        assert value == cfg.read_cache(key)
+
+
+class TestReadCache(TestCase):
+
+    @params(*KNOWN_DATA_PATHS)
+    def test_read_cache_returns_none_when_data_path_absent(
+            self, key, path_basename):
+        """The data_path method returns the data_dir when key is absent."""
+        cfg = UAConfig({'data_dir': '/my/dir'})
+        assert None is cfg.read_cache(key)
+        assert False is os.path.exists(os.path.join('/my/dir', path_basename))
+
+    @params(*KNOWN_DATA_PATHS)
+    def test_read_cache_returns_content_when_data_path_present(
+            self, key, path_basename):
+        tmp_dir = self.tmp_dir()
+        cfg = UAConfig({'data_dir': tmp_dir})
+        data_path = self.tmp_path(path_basename, tmp_dir)
+        with open(data_path, 'w') as f:
+            f.write('content%s' % key)
+
+        assert 'content%s' % key == cfg.read_cache(key)
+
+    @params(*KNOWN_DATA_PATHS)
+    def test_read_cache_returns_stuctured_content_when_json_data_path_present(
+            self, key, path_basename):
+        tmp_dir = self.tmp_dir()
+        cfg = UAConfig({'data_dir': tmp_dir})
+        data_path = self.tmp_path(path_basename, tmp_dir)
+        expected = {key: 'content%s' % key}
+        with open(data_path, 'w') as f:
+            f.write(json.dumps(expected))
+
+        assert expected == cfg.read_cache(key)
+
+
+class TestDeleteCache(TestCase):
+
+    @params('_contracts', '_entitlements', '_machine_token')
+    def test_delete_cache_unsets_any_local_cached_class_attributes(self, attr):
+        """The delete_cache unsets any locally cached class attributes."""
+        tmp_dir = self.tmp_dir()
+        cfg = UAConfig({'data_dir': tmp_dir})
+        assert None is getattr(cfg, attr)
+        setattr(cfg, attr, attr)
+        assert None is not getattr(cfg, attr)
+        assert None is cfg.delete_cache()
+        assert None is getattr(cfg, attr)
+
+    def test_delete_cache_removes_any_cached_data_path_files(self):
```

Comment:
Along the lines of above, I believe this _is_ an appropriate test; we want to be sure that the old values aren't sitting on-disk for our code to pick up in future.

Reply:
+1